src/main/java/org/elasticsearch/index/mapper/object/RootObjectMapper.java

```diff
@@ -176,7 +176,11 @@ protected boolean processField(ObjectMapper.Builder builder, String fieldName, O
                         throw new MapperParsingException("A dynamic template must be defined with a name");
                     }
                     Map.Entry<String, Object> entry = tmpl.entrySet().iterator().next();
-                    ((Builder) builder).add(DynamicTemplate.parse(entry.getKey(), (Map<String, Object>) entry.getValue()));
+                    if(entry.getValue() != null && !((Map<String, Object>) entry.getValue()).isEmpty()) {
+                        ((Builder) builder).add(DynamicTemplate.parse(entry.getKey(), (Map<String, Object>) entry.getValue()));
+                    } else {
+                        ((Builder) builder).add(new DynamicTemplate(entry.getKey(), null, null, null, null, null, null, null, null));
```

Comment:
Can you provide an empty map for the conf instead of null so that we don't need to check for null in thte DynamicTemplate constructor?